### ISSUE TITLE
Android mobile Keyboard interpreted as "shift-enter"

### ISSUE THREAD

**Issue by Ra0R** [jcubic/jquery.terminal#693]
### Issue summary
Happens for me on firefox on android.

### Expected behavior
Command is entered and executed.

### Actual behavior
Command is entered, when pressing "Enter" on the android keyboard a newline character is made but the command is not executed. 

After pressing "Enter" a second time, the command is executed (e.g. "help ", note the newline character after "help")

Any ideas? Might be specific to my implementation.

**Comment by jcubic** [comment]
I need to investigate, will test firefox on Android and let you know. Actually, I didn't test on FF/Android, maybe it works a little bit differently than mobile Chrome that I usually test.

**Comment by jcubic** [comment]
It seems this works fine. Here is my testing page: https://terminal.jcubic.pl/android.php
There is an issue with Firefox calculating the size of the terminal, the terminal is bigger than the width of the browser window. I need to investigate this.

**Comment by jcubic** [comment]
I think the page was zoomed somehow. Terminal works fine on Android in Firefox.

You need to check your settings and/or your code. Maybe you have enabled capslock.

**Comment by jcubic** [comment]
Another idea is that you're using some non-standard keyboard that behaves differently. It would be nice if you can create reproduction. Saying I have an issue without giving any clue what you exactly have is not very helpful. You need a little bit of work otherwise I will not be able to help you.

**Comment by Ra0R** [comment]
Hey there, sorry I didnt get any notifications.
I'm using Gboard which is the Google Keyboard.
The behaviour is the same in the mobile test you linked me.

**Comment by Ra0R** [comment]
![Screenshot_2021-09-29-21-30-30-33_3aea4af51f236e4932235fdada7d1643](https://user-images.githubusercontent.com/22443382/135335957-ec6a79f8-52fd-47fd-898b-9832d3203a43.jpg)

**Comment by Ra0R** [comment]
However it does not seem to be persistent.

**Comment by jcubic** [comment]
I personally use a swift keyboard, I didn't use a built-in keyboard for a while. I have different behavior on that keyboard, I have completion and I need to type enter two times, first to end completion and second to send enter. But I don't have any shift happening.

I have some debugging tools in this terminal like the `keypress` command but I think that I will need a keydown event for this.

Can you try like 10 commands, maybe there is a pattern like each odd is giving shift.
Also what browser do you use Is it Firefox or Chrome or maybe a built-in browser that is kind of different from Chrome?

**Comment by Ra0R** [comment]
So I tried around alot, I feel like it has something to do with the autocomplete function. For instance you press "H", "Helper" is autocompleted and you delete the last 2 letters. 

Maybe autocmplete triggers a keypress/keydown event?

**Comment by Ra0R** [comment]
Some resources I found on this:
https://stackoverflow.com/questions/14194247/key-event-doesnt-trigger-in-firefox-on-android-when-word-suggestion-is-on
https://stackoverflow.com/questions/865490/firefox-capture-autocomplete-input-change-event/6838023

**Comment by jcubic** [comment]
Yes, autocomplete triggers but I think it triggers input event (this is how it use to work when I first started to work on mobile).
On textarea solution, I've disabled autocomplete, but for the Android that I was using it was not working, later the browser added support.

Maybe I just need to disable autocomplete for mobile as well as desktop. But I'm not sure if the browser doesn't ignore this setting.

**Comment by jcubic** [comment]
It should be fixed now, I'm not exactly sure how CM6 fixed the issue but I've figured out this on my own. You can test it on https://terminal.jcubic.pl/android.php (that uses devel branch code).

**Comment by jcubic** [comment]
I'm closing this because of inactivity the issue was fixed and released in version 2.30.0.
Please comment when the bug is still present, I will reopen it.

**Comment by adamclement-exe** [comment]
underscore character on every android keyboard i've tried is interpreted as pressing enter.
tested on my site at http://adamsbashgame.pythonanywhere.com/ as well as  https://terminal.jcubic.pl/android.php
try typing the command "cd Dimension_C-137" or "test_underscore" to replicate the issue

**Comment by jcubic** [comment]
@adamclement-exe I cannot reproduce this on my Phone and BrowserStack, what kind of keyboard you're using?

**Comment by adamclement-exe** [comment]
gboard, swiftkey and samsung keyboards on android

**Comment by jcubic** [comment]
I use swiftkey and I don't see any issues, I will try to switch to gboard.
BTW: Your website is missing the mobile viewport meta tag.
